source/simulator/helper_functions.cc

```diff
@@ -1019,6 +1053,64 @@ namespace aspect
 
         vector.compress(VectorOperation::add);
       }
+    else
+      {
+        // Locally conservative with or without direct solver and with or
+        // without melt: grab a pickaxe and do everything by hand!
```

Comment:
I guess I am not really the right person to look at this part, maybe @bangerth can have a look...
But did you also do a test with melt and DG? Or is that something that doesn't work?